projects/aca-content/src/lib/components/search/search-results/search-results.component.ts

```diff
@@ -189,17 +201,30 @@ export class SearchResultsComponent extends PageComponent implements OnInit {
     this.columns = this.extensions.documentListPresets.searchResults || [];
 
     if (this.route) {
-      this.route.params.forEach((params: Params) => {
-        // eslint-disable-next-line no-prototype-builtins
-        this.searchedWord = params.hasOwnProperty(this.queryParamName) ? params[this.queryParamName] : null;
-        if (this.searchedWord) {
-          this.queryBuilder.update();
+      this.route.queryParams.pipe(takeUntil(this.onDestroy$)).subscribe(async (params: Params) => {
+        this.encodedQuery = params[this.queryParamName] ? params[this.queryParamName] : null;
+        this.searchedWord = extractSearchedWordFromEncodedQuery(this.encodedQuery);
+        this.updateUserQuery();
+        const filtersFromEncodedQuery = extractFiltersFromEncodedQuery(this.encodedQuery);
+        if (filtersFromEncodedQuery !== null) {
+          const filtersToLoad = this.queryBuilder.categories.length;
+          let loadedFilters = this.searchedWord === '' ? 0 : 1;
+          this.queryBuilder.filterLoaded
+            .asObservable()
+            .pipe(takeUntil(this.onDestroy$), takeUntil(this.loadedFilters$))
+            .subscribe(async () => {
+              loadedFilters++;
+              if (filtersToLoad === loadedFilters) {
+                this.loadedFilters$.next();
+                await this.queryBuilder.execute(false);
```

Comment:
Is it necessary to have await here and below? It doesn't seem we are using the return value, or executing any dependent code after queryBuilder.execute